distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso

```diff
@@ -937,13 +951,25 @@ type Table
                  table = Examples.inventory_table
                  double_inventory = table.at "total_stock" * 2
                  table.set "total_stock" double_inventory
-    set : Text -> Column | Vector.Vector -> Table
-    set self name column = case column of
-        _ : Vector.Vector ->
-            self.set name (Column.from_vector name column)
+    set : Text -> Column | Vector.Vector | Text -> Problem_Behavior -> Table
+    set self name column on_problems=Report_Warning = case column of
+        _ : Text ->
+            expression = self.evaluate column
+            if expression.is_error.not then self.set name expression on_problems else
+                on_problems.handle_errors expression fallback=self
+        _ : Vector.Vector -> self.set name (Column.from_vector name column)
         Column.Column_Data _ ->
             Table.Table_Data (self.java_table.addOrReplaceColumn (column.rename name . java_column))
 
+    ## PRIVATE
+    evaluate : Text -> Column
+    evaluate self expression =
+        get_column name = self.at name
+        make_constant value = case value of
+            _ : Column -> value
+            _ -> Column.from_vector (UUID.randomUUID.to_text) (Vector.new self.row_count _->value)
```

Comment:
I think I don't understand this implementation.

The docs for `Expression.evaluate`'s `make_column` say:
```
make_column: a function that takes an object and returns a constant Column object.
```

But here it seems that apart form a constant, this function may get a column, in which case it just returns it as-is. This was not accounted for in the docs for `Expression.evaluate`, so if I was implementing some other code using it I wouldn't know that this is the desired behaviour. 

OTOH, Database does not do this part of logic - so if this behaviour is exceptional only for in-memory backend - then I'd appreciate an explanation why. 

Or maybe it is a leftover that is not needed anymore?

Reply:
Left over from when I couldn't detect an Enso type properly